frontend/src/status-check.ts

```diff
@@ -0,0 +1,55 @@
+interface StatusResponse {
+	deploymentReady: boolean;
+	redirectUrl: string;
+	isError: boolean;
+	errorMessage: string;
+}
+
+const deploymentUrl = new URLSearchParams(location.search).get("redirectTo") || "";
+
+function checkDeploymentStatus() {
+	if (deploymentUrl === "") {
+		return;
+	}
+
+	fetch(`/api/status?deploymentUrl=${deploymentUrl}`, {method: "GET"})
+		.then(response => {
+				if (response.status > 500) {
+					console.log("failed to get deployment status");
+					retry();
+					return null;
+				}
+				return response.json()
+			}
+		).then((response: StatusResponse) => {
+			if (response == null) {
+				return;
+			}
+
+			if (response.isError) {
+				console.log("failed to check deployment status: " + response.errorMessage);
+				retry();
+				return;
+			}
+
+			if (response.deploymentReady) {
+				redirectToDeployment();
+				return;
+			}
+
+			retry();
+		}
+	);
+}
+
+function retry() {
+	setTimeout(() => checkDeploymentStatus(), 1000);
+}
+
+function redirectToDeployment() {
+	window.location.replace(deploymentUrl);
```

Comment:
I was wondering whether we might wan't to update the UI before doing this.
E.g. remove the loading spinner and show a message like "deployment is ready, catnip will redirect you now" and after a brief moment the user get's to the new page.
This might be clearer if the deployed version is somewhat unusable (it usually is a development snapshot after all). Cause now if the deployment is launched but somewhat dysfunctional, catnip just shows the loading spinner and the page just disappears.
I'm not sure though, how realistic the problem scenario is.

Reply:
Done.